spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -51,7 +52,7 @@ class SparkBatch implements Batch {
       JavaSparkContext sparkContext,
       Table table,
       SparkReadConf readConf,
-      List<CombinedScanTask> taskGroups,
+      List<? extends ScanTaskGroup<? extends ScanTask>> taskGroups,
```

Comment:
Methods for planning task groups that preserve grouping work with `ScanTaskGroup` directly.
Also, `BatchScan` returns `ScanTaskGroup`. `CombinedScanTask` extends `ScanTaskGroup `.